extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/VarBinaryChunkInputStreamGenerator.java

```diff
@@ -361,7 +361,8 @@ public int drainTo(final OutputStream outputStream) throws IOException {
             final MutableInt logicalSize = new MutableInt();
             subset.forAllRowKeys((idx) -> {
                 try {
-                    logicalSize.add(byteStorage.getPayloadSize((int) idx, (int) idx));
+                    logicalSize.add(LongSizedDataStructure.intSize("int cast",
```

Comment:
As with the row count for batches (and so the size of the buffers that hold the data for a given column), a given value in a column is also bounded by the size that it would be when read/written by languages that don't deal with arrays bigger than 2^32. For example, java strings, bigints, bigdecimals, or arrays of bytes will be written out using this class, so the payload must not be longer than what a Java array can be sized.